tests/versioned/express-segments-tests.js

```diff
@@ -0,0 +1,661 @@
+          author {
+            name
+          }
+        }
+      }
+    }`
+
+    const longestPath = 'libraries.books.author.name'
+
+    helper.agent.on('transactionFinished', (transaction) => {
+      const operationPart = `query ${ANON_PLACEHOLDER}`
+      const expectedSegments = [{
+        name: `WebTransaction/apollo-server/${operationPart} ${longestPath}`,
+        children: [{
+          name: 'Expressjs/Router: /',
+          children: [{
+            name: 'Nodejs/Middleware/Expressjs/<anonymous>',
+            children: [{
+              name: operationPart,
+              children: [
+                { name: 'resolve: libraries' },
+                { name: 'resolve: libraries.books' },
+                { name: 'resolve: libraries.books.author' },
+                { name: 'resolve: libraries.books.author.name' }
+              ]
+            }]
+          }]
+        }]
+      }]
+
+      t.segments(transaction.trace.root, expectedSegments)
+    })
+
+    executeQuery(serverUrl, query, (err, result) => {
+      t.error(err)
+      checkResult(t, result, () => {
+        t.end()
+      })
+    })
+  })
+
+  t.test('named query, multi-level should return longest path', (t) => {
+    const { helper, serverUrl } = t.context
+
+    const expectedName = 'GetBooksByLibrary'
+    const query = `query ${expectedName} {
+      libraries {
+        books {
+          title
+          author {
+            name
+          }
+        }
+      }
+    }`
+
+    const longestPath = 'libraries.books.author.name'
+
+    helper.agent.on('transactionFinished', (transaction) => {
+      const operationPart = `query ${expectedName}`
+      const expectedSegments = [{
+        name: `WebTransaction/apollo-server/${operationPart} ${longestPath}`,
+        children: [{
+          name: 'Expressjs/Router: /',
+          children: [{
+            name: 'Nodejs/Middleware/Expressjs/<anonymous>',
+            children: [{
+              name: operationPart,
+              children: [
+                { name: 'resolve: libraries' },
+                { name: 'resolve: libraries.books' },
+                { name: 'resolve: libraries.books.title'},
+                { name: 'resolve: libraries.books.author' },
+                { name: 'resolve: libraries.books.author.name' }
+              ]
+            }]
+          }]
+        }]
+      }]
+
+      t.segments(transaction.trace.root, expectedSegments)
+    })
+
+    executeQuery(serverUrl, query, (err, result) => {
+      t.error(err)
+      checkResult(t, result, () => {
+        t.end()
+      })
+    })
+  })
+
+  t.test('named query, multi-level, should choose *first* deepest-path', (t) => {
+    const { helper, serverUrl } = t.context
+
+    const expectedName = 'GetBooksByLibrary'
+    const query = `query ${expectedName} {
+      libraries {
+        books {
+          title
+          isbn
+        }
+      }
+    }`
+
+    // .isbn is the same length but title will be first so that path should be used
```

Comment:
.isbn?

Reply:
one of the queried fields... so there will be two paths of equal length [*.title, *.isbn]